proposals/VBnet-Coding-Guidelines.md

```diff
@@ -0,0 +1,78 @@
+# VB.net Coding Guidelines
+ 
+ Coding Conventions
+ 
+ * 80 character line limit    
```

Comment:
❗️ There are two issues with this:

1. There is no real line length limit in place
2. If there was a line length limit, it wouldn't be 80 characters

I wouldn't be against mentioning a soft limit of 120 characters or more.

Reply:
Having both an 80 character line limit and 4 character "tab" seems, to me, to conflict a bit... a lot of code could easily have 5 or more "tabs"...

Namespace...
.Class...
..Function...
...Select Case...
....Case...
.....If...

So 1/4th of the line is eaten up with tabs.

Ouch.

Reply:
@DualBrain I know I personally prefer 2 space character tab, but the `/dotnet/...` prefers 4.
@sharwell My rationale behind 80 character was tradition, but I am ok with 120.

The issue / pr was the stimulate discussion, and tempt the team on producing a preferred style for usage in the repos.  *As I was clashing heads, when submit pull-request due the lack of them*

Reply:
@AdamSpeight2008 I would think that the coding guidelines would be based on the language, not to have to adhere to arbitrary limitations decided because of inherit "limitations" of another language.  I think VB should have 2 character "tabs"... having 4 in C# makes sense given the inherit lack of clarity between start/end constructs.  It makes perfect sense to want to have more spacing... it provides a clear differentiation.  VB on the other hand has clear constructs... thus the spacing of 4 is kind of ridiculous.  I also agree that the line limit should be increased... 120 seems more reasonable (especially if 2 spaces were to be utilized).  Again, lines in VB are long... it has greater verbosity.